tests/test_core.py

```diff
@@ -162,10 +188,21 @@ def test_region_processing_complete(directory):
                 ["model_a", "scen_a", "World", "Price|Carbon", "USD/t CO2", 2.8, 7.0],
                 ["model_a", "scen_a", "World", "Price|Carbon (Max)", "USD/t CO2", 3, 8],
             ],
+            None,
+        ),
+        # check that region-aggregation with missing weights passes (inconsistent index)
+        # TODO check the log output
+        (
+            "weighted_aggregation",
+            [
+                ["model_a", "scen_a", "World", "Primary Energy", "EJ/yr", 4, 6],
+                ["model_a", "scen_a", "World", "Emissions|CO2", "Mt CO2", 5, np.nan],
```

Comment:
Just a question for my understanding. The filtering dictionary which removes the 2010 entry for Emissions|CO2 is needed because directly creating a data frame with the 2010 data missing for one variable would throw an error, right?

Reply:
No, the filtering applies to `test_df`. This is to use the same test-data for all different versions of this test. Removing this one value creates an inconsistent index.